src/Statistics/views.py

```diff
@@ -45,4 +53,342 @@ def durationPlay(request):
+    stats = StatSinglePlay.objects.all()
+    user = request.user
+    fiveBestSongs = findBestFiveSongs(findMostPopularUserSongs(user))
+    fiveBestArtists = findBestFiveSongs(findMostPopularUserArtists(user))
+    fiveBestGenres = findBestFiveSongs(findMostPopularUserGenres(user))
+
+    context = {'stats': stats, 'fiveBestSongs': fiveBestSongs, 'fiveBestArtists': fiveBestArtists, 'fiveBestGenres': fiveBestGenres}
+    return render(request, 'Statistics/user_stats.html', context=context)
+
+
+def adminStats(request):
+    return render(request, 'Statistics/admin_stats.html', context={})
+
+
+def findNDifferentColors(n):
+    colors = []
+
+    while len(colors) < n:
+        color = "#" + ''.join([random.choice('0123456789ABCDEF') for _ in range(6)])
+        if color not in colors:
+            colors.append(color)
+
+    return colors
+
+
+def createPDF():
+    pdf = FPDF('L', 'mm', 'A4')
+    pdf.add_page()
+    pdf.set_font('courier', 'B', 20)
+
+    return pdf
+
+
+def addToPDF(pdf, data, title, tableTitlesData, tableContentData):
+    pdf.set_font('courier', 'B', 20)
+    pdf.cell(297, 10, title, ln=1, align='C')
+    pdf.cell(297, 10, '', ln=1)
+    pdf.set_font('courier', 'B', 14)
+
+    if len(tableTitlesData) == 5:
+        pdf.cell(297, 14, f" {tableTitlesData[0].ljust(tableTitlesData[1])} {tableTitlesData[2].ljust(tableTitlesData[3])} {tableTitlesData[4]}", ln=1)
+    elif len(tableTitlesData) == 7:
+        pdf.cell(297, 14, f"{tableTitlesData[0].ljust(tableTitlesData[1])} {tableTitlesData[2].ljust(tableTitlesData[3])} "
+                          f"{tableTitlesData[4].ljust(tableTitlesData[5])}  {tableTitlesData[6]}", ln=1)
+
+    pdf.line(10, 40, 287, 40)
+    pdf.set_font('courier', '', 12)
+    data1, data2, data3 = [], [], []
+    idx = 1
+
+    if len(tableTitlesData) == 5:
+        for el1, el2 in data:
+            pdf.cell(297, 7, str(idx).ljust(tableContentData[0]) + str(el1).ljust(tableContentData[1]) + str(el2).rjust(tableContentData[2]), ln=1)
+            data1.append(el1)
+            data2.append(el2)
+            idx += 1
+
+    elif len(tableTitlesData) == 7:
+        for el1, el2, el3 in data:
+            pdf.cell(297, 7, str(idx).ljust(tableContentData[0]) + str(el1).ljust(tableContentData[1]) +
+                     str(el2).ljust(tableContentData[2]) + str(el3).rjust(tableContentData[3]), ln=1)
+            data1.append(el1)
+            data2.append(el2)
+            data3.append(el3)
+            idx += 1
+
+    return data1, data2, data3, pdf
+
+
+def addContentToMyReport(pdf, data, tableContentData, title, tableTitlesData, xLabel, yLabel, id_):
+    data1, data2, data3, pdf = addToPDF(pdf, data, title, tableTitlesData, tableContentData)
+    x = [str(i) for i in range(1, len(data1) + 1)]
+
+    if len(tableTitlesData) == 3:
+        createBarPlot(x, data2, title, xLabel, yLabel, id_)
+    elif len(tableTitlesData) == 7:
+        createBarPlot(x, data3, title, xLabel, yLabel, id_)
+
```

Comment:
Incorrect length of _tableTitlesData_ causes image not to be created; unhandled exception.